pennylane/templates/subroutines/qsvt.py

```diff
@@ -464,3 +466,290 @@ def _qsp_to_qsvt(angles):
+
+        It is important to note the differences between QSP and GQSP:
+
+        1. Given a block encoding of a matrix :math:`A`, GQSP works on the Chebyshev basis and QSP on the standard basis.
+        2. QSP accepts as input only polynomials with defined parity but GQSP does not have this restriction.
+        3. In GQSP, the coefficients can be complex, while in QSP they must be real.
+        4. GQSP applies three parameters per iteration, while QSP applies only one.
+
+        Based on [1], in order to calculate the angles for a polynomial :math:`F` in QSP using a GQSP solver, the
+        polynomial should first be transformed into the Chebyshev basis. This is done via `chebyshev.poly2cheb(F)`.
+
+        Theorem 5 of the paper [`arXiv:2406.04246 <https://arxiv.org/abs/2406.04246>`_] shows that to calculate the complementary polynomial of :math:`F`, we
+        first define :math:`P(z) = z^{\frac{d}{2}}F(\sqrt{z})`. Based on [2], :math:`F` is of the
+        form :math:`(a_0, 0, a_2, 0, a_4, \dots)`, so one can
+        rewrite :math:`P` as :math:`(0, 0, \dots, a_0, a_2, a_4, \dots)`. This is what is done
+        when :math:`P` is defined in the code.
+
+        Once we have the coefficients of the polynomial and its complementary,
+        we can apply Algorithm 1 in [`arXiv:2308.01501 <https://arxiv.org/abs/2308.01501>`_], where :math:`\vec{a}` and :math:`\vec{b}` are the
+        coefficients of these polynomials, respectively. By [3], the coefficients are real,
+        so the :math:`\vec{\theta}`, :math:`\vec{\phi}`, and :math:`\vec{\lambda}` angles can be combined,
+        thus generating ``rotation_angles``. The algorithm is simplified, and instead of working
+        with the gate defined in (eq.8) of the previous paper, the RY gate is used.
+    """
+
+    parity = (len(F) - 1) % 2
+
+    # Construct the auxiliary polynomial P and its complementary Q based on appendix A in [arXiv:2406.04246]
+    P = np.concatenate([np.zeros(len(F) // 2), chebyshev.poly2cheb(F)[parity::2]]) * (1 - 1e-12)
+    complementary = _complementary_poly(P)
+
+    polynomial_matrix = np.array([P, complementary])
+    num_terms = polynomial_matrix.shape[1]
+    rotation_angles = np.zeros(num_terms)
+
+    # This subroutine is an adaptation of Algorithm 1 in [arXiv:2308.01501]
+    # in order to work in the context of QSP.
+    with qml.QueuingManager.stop_recording():
+        for idx in range(num_terms - 1, -1, -1):
+
+            poly_a, poly_b = polynomial_matrix[:, idx]
+            rotation_angles[idx] = np.arctan2(poly_b.real, poly_a.real)
+
+            rotation_op = qml.matrix(qml.RY(-2 * rotation_angles[idx], wires=0))
+
+            updated_poly_matrix = rotation_op @ polynomial_matrix
+            polynomial_matrix = np.array(
+                [updated_poly_matrix[0][1 : idx + 1], updated_poly_matrix[1][0:idx]]
+            )
+
+    return rotation_angles
+
+
+def transform_angles(angles, routine1, routine2):
```

Comment:
How the transformation works? We need either a proper citation or some explanation in the doctsring (maybe under a Theory or Details section).